include/swift/AST/ParseRequests.h

```diff
@@ -99,6 +100,9 @@ class ParseSourceFileRequest
   bool isCached() const { return true; }
   Optional<ArrayRef<Decl *>> getCachedResult() const;
   void cacheResult(ArrayRef<Decl *> decls) const;
+
+public:
+  evaluator::DependencySource readDependencySource(Evaluator &) const;
```

Comment:
I would not expect parsing to register dependency edges since it should not be doing name lookup or anything else. Do we have a way to assert that?

Reply:
Not that I know of. We could potentially throw a "no request evaluation" scope around the parser...